pkg/daemon/update.go

```diff
@@ -162,8 +163,35 @@ func (dn *Daemon) drainAndReboot(newConfig *mcfgv1.MachineConfig) (retErr error)
 
 var errUnreconcilable = errors.New("unreconcilable")
 
+func canonicalizeEmptyConfig(config *mcfgv1.MachineConfig) *mcfgv1.MachineConfig {
```

Comment:
this could be simpler by just having a canonicalizeMC() and instead of only returning the empty config also doing a check to see if the kargs slice is empty and if so, making it a nil slice. if you did this here, you wouldn't need the other helper below - you'd just use this on both the oldConfig and newConfig immediately in update()

Reply:
I would prefer not to change anything with an existing config. Empty slice and nil being different becomes a problem only in NewMachineConfigDiff (because it uses DeepEqual), so ideally the fix should be added there itself.

Reply:
Can we change this to canonicalizeEmptyMC? just to indicate it is a machineconfig?

Reply:
SGTM, renamed!